### PR TITLE
docs(changelog): add changelog for version 5.1.0 to 3.0.0

### PR DESCRIPTION
**Pull request by ChristianMurphy**
:information_source: About Keep a Changelog
This takes a different approach to keeping a changelog.
Rather than a primarily automated process based off commit messages. :robot: 
This takes the purely human approach of letting the release manager add a description of changes. :memo: 

While still offering some recommendations on how to structure the changelog.

:link: from [Keep a Changelog docs](http://keepachangelog.com/en/1.0.0/)

#### Guiding Principles

*  Changelogs are *for humans*, not machines.
* There should be an entry for every single version.
* The same types of changes should be grouped.
*  Versions and sections should be linkable.
*  The latest version comes first.
* The release date of each versions is displayed.
* Mention whether you follow [Semantic Versioning](http://semver.org/).

#### Types of changes

* `Added`  for new features.
* `Changed`  for changes in existing functionality.
* `Deprecated` for soon-to-be removed features.
* `Removed` for now removed features.
* `Fixed` for any bug fixes.
* `Security` in case of vulnerabilities.

:construction: Code changes

Copies release notes from GitHub releases 3.0.0 to present into `CHANGLOG.md`.

----

Contributor License Agreement adherence:

- [x] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+## [3.1.4][] - 2017-03-20
+
+Fixes a bug where text in a static portlet was hard to read (#380)
+
+## [3.1.3][] - 2017-03-09
+
+Adds back in UW-Lacrosse theming
+
+## 3.1.2 - \[Yanked]
```
Comment (ChristianMurphy) [inline CHANGELOG.md]: :question: any details on why this release was skipped or yanked that could be added?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+Patch release. Trivial upgrade from `v3.0.1` (or even `v3.0.0`).
+- Username menu items are no longer redundantly duplicatively labeled ( #351 )
+- Circle buttons now maintain their circular shape even when their label text wraps ( #352 )
+
+Supporting links:
+- [Documentation for this version specifically](http://uw-madison-doit.github.io/uw-frame/v3.0.2/).
+- [Release announcement on myuw-developers group](https://groups.google.com/d/topic/myuw-developers/tKOWjo8r96c/discussion)
+
+## 3.0.1 - \[YANKED]
```
Comment (ChristianMurphy) [inline CHANGELOG.md]: :question: any details on why this release was skipped or yanked that could be added?

File: CHANGELOG.md (hunk 1 above)
Comment (vertein) [reply CHANGELOG.md]: My guess is that something went wrong with the release to Maven central, so the git tag was yanked, and the release process tried again.

I'm not 100% sure, but I'm 95% confident that's what happened.

File: CHANGELOG.md (hunk 2 above)
Comment (vertein) [reply CHANGELOG.md]: My guess is that something went wrong with the release to Maven central, so the git tag was yanked, and the release process tried again.

I'm not 100% sure, but I'm 95% confident that's what happened.

Comment (ChristianMurphy) [reply CHANGELOG.md]: Thanks @vertein, I updated the changelog with the correct date for yanked releases as well as a note on maven central issues.

**Pull request merged**